clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractStickyAssignor.java

```diff
@@ -43,7 +43,7 @@
 
     public static final int DEFAULT_GENERATION = -1;
 
-    private PartitionMovements partitionMovements = new PartitionMovements();
+    private PartitionMovements partitionMovements;
```

Comment:
Can we still initialize it here as well? I remember that was necessary for some tests to pass since they might never get to the `generalAssign` method and `isSticky` would hit NPE

On the other hand, it seems like `isSticky` is pointless to call unless we get to the `generalAssign` method. So maybe we should just remove that from the tests that only do the `constrainedAssign` and just verify the stickiness directly?